lib/passport-configurator.js

```diff
@@ -307,232 +307,231 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
             }
           }
-          return res.redirect(successRedirect(req));
+          return res.redirect( successRedirect( req ) );
+        } );
+      } else {
+        if ( info && info.accessToken ) {
+          if ( !!options.json ) {
+            return res.json( info.accessToken );
+          } else {
+            res.cookie( 'access_token', info.accessToken.id, {
+              signed: req.signedCookies ? true : false,
+              maxAge: 1000 * info.accessToken.ttl,
+            } );
+            res.cookie( 'userId', user.id.toString(), {
+              signed: req.signedCookies ? true : false,
+              maxAge: 1000 * info.accessToken.ttl,
+            } );
+          }
         }
-      })(req, res, next);
+        return res.redirect( successRedirect( req ) );
+      }
+    } )( req, res, next );
   };
   /*
    * Redirect the user to Facebook for authentication.  When complete,
    * Facebook will redirect the user back to the application at
    * /auth/facebook/callback with the authorization code
    */
-  if (authType === 'local') {
-    self.app.post(authPath, passport.authenticate(
-      name, options.fn || _.defaults({
-        successReturnToOrRedirect: options.successReturnToOrRedirect,
-        successRedirect: options.successRedirect,
-        failureRedirect: options.failureRedirect,
-        successFlash: options.successFlash,
-        failureFlash: options.failureFlash,
-        scope: scope, session: session,
-      }, options.authOptions)));
-  } else if (authType === 'ldap') {
+  if ( authType === 'local' ) {
+    self.app.post( authPath, (options.callbackFn ? options.callbackFn( passport ) :
+        passport.authenticate(
+          name, options.fn || _.defaults({
+            successReturnToOrRedirect: options.successReturnToOrRedirect,
+            successRedirect: options.successRedirect,
+            failureRedirect: options.failureRedirect,
+            successFlash: options.successFlash,
+            failureFlash: options.failureFlash,
+            scope: scope, session: session
+          }, options.authOptions)
+        )
+    ));
+  } else if ( authType === 'ldap' ) {
     var ldapCallback = options.customCallback || defaultCallback;
-    self.app.post(authPath, ldapCallback);
-  } else if (link) {
-    self.app.get(authPath, passport.authorize(name, _.defaults({
-      scope: scope,
-      session: session,
-    }, options.authOptions)));
+    self.app.post( authPath, ldapCallback );
+  } else if ( link ) {
+    self.app.get(authPath, (options.callbackFn ? options.callbackFn( passport, options ) :
+      passport.authorize(name, _.defaults({
+        scope: scope,
+        session: session,
+      }, options.authOptions))));
   } else {
-    self.app.get(authPath, passport.authenticate(name, _.defaults({
-      scope: scope,
-      session: session,
-    }, options.authOptions)));
+    self.app.get(authPath, (options.callbackFn ? options.callbackFn( passport, options ) :
```

Comment:
so it looks like this is allowing you to setup a custom passport auth handler. Is there a better name than just `callbackFn`? Also it looks like this is still backwards compatible so this would be a feature version bump.